### PR TITLE
[TASK] Add `php.ini` from development for GitLab CI

### LINKED ISSUE #1214: Use the development PHP INI on CI in gitlab CI
## Goal
I as a user would like to see the PHP errors, warnings and notices in the CI.

## Acceptance Criteria
- [x] the development php.ini is used

## Additional Information
This will allow us to see (and error out on) all warnings and notices.
https://github.com/shivammathur/setup-php/issues/827
Related: #1197

### PR DESCRIPTION
**Pull request by linxpinx**
Fixed #1214

### CODE REVIEW

File: .gitlab/pipeline/ci/php.ini [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+[Pdo_mysql]
+pdo_mysql.cache_size = 2000
+pdo_mysql.default_socket=
+
+[mail function]
+SMTP = localhost
+smtp_port = 25
+mail.add_x_header = On
+sendmail_path="/usr/local/bin/mailpit sendmail -t --smtp-addr 127.0.0.1:1025"
```
Comment (oliverklee) [inline .gitlab/pipeline/ci/php.ini:73]: We're not using Mailpit in GitLab CI (and also not on GitHub Actions), so this line doesn't look right.

File: .gitlab/pipeline/jobs/.default.yml [hunk 2]
```diff
@@ -2,3 +2,7 @@
   image: ghcr.io/sbuerk/demo-core-testing-php83:latest
   before_script:
     - bash .gitlab/build/docker_install.sh > /dev/null
+  variables:
+    PHP_INI_SCAN_DIR: "/etc/php"
+  volumes:
+    - ../ci/php.ini:/usr/local/etc/php/php.ini
```
Comment (oliverklee) [inline .gitlab/pipeline/jobs/.default.yml:8]: There should always be a trailing LF after the last line in a file.

**Review by oliverklee — CHANGES_REQUESTED** [review]
With this, the CI build is failing:

jobs:build-composer-dependencies config contains unknown keys: volumes

https://git.typo3.org/qa/example-extension/-/pipelines/66001

**Comment by linxpinx** [comment]
I have updated the PR can you review again? @oliverklee @lukaszuznanski

**Comment by linxpinx** [comment]
I had to close this PR cause only PR created after 1.10.2024 and created with my Github Account will be seen by Hacktober.com

**Pull request closed without merging**